Providers/nxOMSAutomationWorker/automationworker/3.x/worker/jrdsclient.py

```
@@ -79,10 +88,82 @@ def get_sandbox_actions(self):
+        """ Rotate worker certificate. 
+        Steps includes creating new certificate and after Server returns 200, replace the old certificate with newly generated certificate.
+        """
+        import tracer
+        try:
+            temp_certificate_path, temp_key_path = workercertificaterotation.generate_cert_rotation_self_signed_certificate()
+            issuer, subject, thumbprint, not_before, not_after = linuxutil.get_cert_info(temp_certificate_path)
+
+            payload = {'Thumbprint': thumbprint,
+               'Issuer': issuer,
+               'Subject': subject,
+               'NotBefore': not_before,
+               'NotAfter': not_after}
+
+            headers = {"Content-Type": "application/json"}
+
+            url = self.base_uri + "/automationAccounts/" + self.account_id + \
+              "/hybridCertificateRotation?api-version=" + self.protocol_version
+            response = self.issue_request(lambda u: self.httpClient.post(u, headers=headers, data=payload), url)
+
+            if response.status_code == 200:
+                import tracer
+                tracer.log_worker_debug("Worker certificate successfully got rotated on the server side.")
+                workercertificaterotation.replace_self_signed_certificate_and_key(temp_certificate_path, temp_key_path, thumbprint)
+                tracer.log_worker_debug("replace_self_signed_certificate_and_key done")
+                return
+            else:
+                import tracer
+                tracer.log_worker_debug(response)
+                tracer.log_worker_debug(dir(response))
+                tracer.log_worker_debug(response.status_code)
+
+            workercertificaterotation.clean_up_certificate_and_key(temp_certificate_path, temp_key_path)
+
+        except:
+            import tracer
+            tracer.log_worker_debug(traceback.format_exc())
+
+        raise Exception("Hybrid Worker certificate rotation failed. [status=" + str(response.status_code) + "]")
```

do we need to raise exception ?
what will happen if we do and what will we loose if we do not, we can silently continue ?

---

we can log it in worker.log